nixos/modules/config/fonts/fontdir.nix

```diff
@@ -5,14 +5,16 @@ with lib;
 let
 
   x11Fonts = pkgs.runCommand "X11-fonts" { preferLocalBuild = true; } ''
+    font_formats=('ttf' 'otf' 'pcf' 'pfa' 'pfb' 'bdf')
     mkdir -p "$out/share/X11-fonts"
-    find ${toString config.fonts.fonts} \
-      \( -name fonts.dir -o -name '*.ttf' -o -name '*.otf' \) \
-      -exec ln -sf -t "$out/share/X11-fonts" '{}' \;
+    for fmt in "''${font_formats[@]}"; do
```

Comment:
I think you should be able to use `-regex` to avoid calling `find` multiple times and the `-o`.

Reply:
I implemented that at the cost of a more complicated code, I'm open to suggestions for a simpler regexp-based way

Reply:
I amended the commit to simplify a bit more